pkg/clusteragent/custommetrics/provider.go

```diff
@@ -65,7 +65,11 @@ func (p *datadogProvider) ListAllMetrics() []provider.CustomMetricInfo {
 
 // ListAllExternalMetrics is called every 30 seconds, although this is configurable on the API Server's end.
 func (p *datadogProvider) ListAllExternalMetrics() []provider.ExternalMetricInfo {
-	cl := hpa.GetHPAWatcherClient()
+	cl, err := hpa.GetHPAWatcherClient()
+	if err != nil {
+		log.Errorf("Could not get the HPA Client: %s", err.Error())
+		return nil
+	}
```

Comment:
What are the challenges with passing a reference to the `HPAWatcherClient` in `NewDatadogProvider`?

It seems a bit odd to instantiate a type that uses a Start/Stop pattern and never actually Start it.